### DISCUSSION TITLE
Clock

### DISCUSSION THREAD

**Discussion by IHOXOHI** [micropython/micropython#14289] [STM32 / Pyboard] [answered]
Hi,
It seems that the clock of the giga, portenta no works...
If I can do something....
I tried the calibration command; it works fine on a pyboard for example, but it's a precision correction; not enough power to calibrate the giga...
Thanks.

**Comment by IHOXOHI** [comment]
On other hand, I'm writing something, ..., I don't known... I posted on the arduino forum to tell it...
It seems that a ucontroller witha lots of pin, could manage lots of buttons, so make a keyboard.
Until now, my giga can write on his screen his own characters, save varaibles, open/close a file, the print works, the if, for too... I will try the import function... If it works... It's a computer, not a ucontroller... But it's clear that it's better to manage the eval function in the firmware... I presume that it's for security that it's not possible to do an eval('import something')...
![giga1](https://github.com/micropython/micropython/assets/63340434/b985e180-901c-4fcb-9a55-33807424a454)

**Reply by scruss** [reply]
Please read the [Welcome (Please read before posting) · micropython · Discussion #9111](https://github.com/orgs/micropython/discussions/9111) posting so you know how to format code so we can read it

**Comment by IHOXOHI** [comment]
Hey,
Finally, I tested the behaviour of the giga and 'los 2 portenta'; the stability isn't affected.... It's just that the time isn't correctly returned... Why? I don't know... How it can be, 'egal'.
So, for me, I have just add another ucontroller (the pyb405 is fine with his rasperry_hat, a clock battery, sdcard...), and communicate by uart with it to check the time if necessary.
Thanks.

**Reply by IHOXOHI** [reply, marked as answer]
https://github.com/IHOXOHI/IMI/tree/main
